roles/translator/src/proxy/bridge.rs

```diff
@@ -141,57 +144,68 @@ impl Bridge {
     /// Receives a SV1 `mining.submit` message from the `Downstream`, translates it to a SV2
     /// `SubmitSharesExtended` message, and sends it to the `Upstream`.
     fn handle_downstream_share_submission(self_: Arc<Mutex<Self>>) {
-        let rx_sv1_submit = self_.safe_lock(|s| s.rx_sv1_submit.clone()).unwrap();
-        let tx_sv2_submit_shares_ext = self_
-            .safe_lock(|s| s.tx_sv2_submit_shares_ext.clone())
+        let (rx_sv1_submit, tx_sv2_submit_shares_ext, target, tx_status) = self_
+            .safe_lock(|s| {
+                (
+                    s.rx_sv1_submit.clone(),
+                    s.tx_sv2_submit_shares_ext.clone(),
+                    s.target.clone(),
+                    s.tx_status.clone(),
+                )
+            })
             .unwrap();
-        let target = self_.safe_lock(|s| s.target.clone()).unwrap();
-        let tx_status = self_.safe_lock(|s| s.tx_status.clone()).unwrap();
         task::spawn(async move {
             loop {
-                let target = target.safe_lock(|t| t.clone()).unwrap().try_into();
+                let target = target.safe_lock(|t| t.clone()).map_err(|_| PoisonLock);
```

Comment:
We already cloned the target above at 152

Reply:
The one that returns above is an `Arc<Mutex>>`  so it has to be locked again.  Maybe `target_mutex` would be a better name